### ISSUE TITLE
IOS App: Adding Comments does not use the IOS's username but shows LocalHost#xx

### ISSUE THREAD

**Issue by sebastiaanveld** [CollaboraOnline/online#1843]
**Describe the bug**
Adding Comments in documents does not use the IOS's username but shows LocalHost#xx.
Tested this with the lastest App 6.4.7 (1), think this worked in the past but not sure which version.

**To Reproduce**
1. Open a document in Collabora Office on IOS
2. Enabled Edit-mode
3. Add a Comment
4. The commenter-field does not show "<my username>" but "LocalHost#xx"

**Expected behavior**
I would expect the IOS username would be used as available in the linked Apple ID, or maybe an override for that which can be setup in the app preferences in the IOS Settings> Collabora Office section in case the Apple ID name is not the name the user wants to display in the comments.

**Actual behavior**
Seems another field is used for fill the username, resulting in the username displaying "LocalHost#xx" 

**Screenshots**
![afbeelding](https://user-images.githubusercontent.com/37268999/112447051-28b8d000-8d51-11eb-9502-ddbb5b610f6f.png)

**Smartphone**
 - Device: iPhone X / iPad 5th gen
 - OS: IOS 14.4.1
- App: Collabora Office 6.4.7 (1)

**Comment by thebearon** [comment]
Indeed, and it is the same in the Android app.

**Comment by nikslor** [comment]
This is a duplicate of https://bugs.documentfoundation.org/show_bug.cgi?id=123729

@thebearon: should we keep both issues open to avoid further duplicates?
@sebastiaanveld: i have roughly 50 iOS specific issues in BZ; happy to share my list if you'd like to have a look at it (and thanks a looooot for testing as well <3)

**Comment by sebastiaanveld** [comment]
I stopped looking at BZ as this is (for now) the place to report issues:) 

@nikslor maybe add those issues already reported in BZ to a meta bug and refer to that in duplicate issue opened here.  And/or then add a link to the meta bug in the New Bug template text so when opening a new issue one can go and check BZ and then add the BZ issue ID here too. If you close duplicates here they sooner or later just will get recreated. If you want to minimize re-creating duplicated I would suggest to create issues for any of those 50 BZ issues here and link them which may also be the easiest way to track progress.

**Comment by tml1024** [comment]
This is harder than it seems. There is no desktop-like user account concept in iOS.

As far as I understand, an iOS device usually is associated with an Apple ID account, but I think that is optional. Also, there usually is an Apple ID used for App Store purchases, but it does not need to be the same. And in situations like remote managed devices it is unclear whether there is any of these. Access to these Apple IDs, if they exist, and especially the full name of the person(s) in question, is likely not straightforward, if even permitted, from apps. (At least I haven't figured out it yet.)

Then there is Contacts, and many people presumably have themselves in their Contacts database, too, including their full name. (Well, at least I do.) But likely not all. And for the Collabora Office iOS app would request access to Contacts, that would surely be seen as weird by many users. It would have to be mentioned in the App Store listing for the app, and would surely turn many potential users away, "what, why would a document editor need my Contacts, I am not going to install this".

But yeah, investigating more.

**Comment by tml1024** [comment]
FWIW, calling NSFullUserName() in the iOS app gives you "Mobile User", not very useful.

(Apple's documentation for that function does not mention at all that it is useless on iOS, even if it exists.)

**Comment by nikslor** [comment]
@tml1024 I still think the simplest way would be to have a couple of fields in the app settings (the one that iOS offers, we already have the "Template list URL" there, why not adding "Firstname" and "Lastname" there?)

**Comment by tml1024** [comment]
@nikslor Yep, that is not hard. Will do it that way. (Not using separate "first name" and "last name" fields, though; see https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/  ...)

**Comment by tml1024** [comment]
@nikslor Should the user name also be a field that can be set using a mobile device management system, like the templateListURL? (And in that case, this will override whatever the user attempts to put there.)

**Comment by nikslor** [comment]
Yes, setting thigs through MDM would be a big plus. And yes, not having a seprate field is of course good!

**Comment by tml1024** [comment]
Implemented now for iOS, keeping this open for the Android app.
